### PR TITLE
🌱 update grpc option for cloudevent client.

### PR DESCRIPTION
**Pull request by morvencao**
## Summary

This PR tries to update the grpc option for the cloudevent client.
- remove topic from publish and subscription requests
- add source to subscription request

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/cloudevents/generic/options/grpc/agentoptions.go [hunk 1]
```diff
@@ -60,12 +60,11 @@ func (o *grpcAgentOptions) Client(ctx context.Context) (cloudevents.Client, erro
...
-		protocol.WithPublishOption(&protocol.PublishOption{}),
 		protocol.WithSubscribeOption(&protocol.SubscribeOption{
-			Topics: []string{
-				replaceNth(SpecTopic, "+", o.clusterName, 2), // receiving the resources spec from sources with spec topic
-				StatusResyncTopic, // receiving the resources status resync request from sources with status resync topic
-			},
+			// TODO: Update this code to determine the subscription source for the agent client.
+			// Currently, the grpc agent client is not utilized, and the 'Source' field serves
+			// as a placeholder with the cluster name.
+			Source: o.clusterName,
```
Comment (skeeey) [inline pkg/cloudevents/generic/options/grpc/agentoptions.go:67]: can we use `types.SourceAll` here? it is an empty string

Comment (skeeey) [reply pkg/cloudevents/generic/options/grpc/agentoptions.go:67]: BTW. I think you can remove the https://github.com/open-cluster-management-io/sdk-go/blob/main/pkg/cloudevents/generic/options/grpc/options.go#L23 now

Comment (morvencao) [reply pkg/cloudevents/generic/options/grpc/agentoptions.go]: done.

**Comment by skeeey** [comment]
LGTM

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**